cmd/lncli/psbt.go

```diff
@@ -0,0 +1,283 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/signal"
+	"github.com/urfave/cli"
+)
+
+// openChannelPSBT starts an interactive channel open protocol that uses a
+// partially signed bitcoin transaction (PSBT) to fund the channel output. The
+// protocol involves several steps between the RPC server and the CLI client:
+//
+// RPC server                           CLI client
+//     |<----register funding shim----------|
```

Comment:
So here's my core question again: why does the user need to register the funding shim _before_ they open the channel? Compared to just passing it in with the original funding request (RPC level).

Reply:
Other than this, the external RPC flow looks sound IMO.

Reply:
Fixed it by removing the initial register call.